doc/Operation.md

```diff
@@ -0,0 +1,328 @@
+
+Any concrete class of `Operation<T>` will expose a `Poll()` and `PollUntilDone()` APIs. These methods will update the `operation status`. The status of an operation won't be automatically updated without calling `Poll()`.
+
+### Poll
+
+The `Poll()` method will send a request to the Server to get the current status of the `LRO`. The Server will then return an Http raw response with information about the completion of the LRO. The Operation parses the response and updates its status. Finally, a read only reference to the http raw response (the one used to update the status) is returned by `Poll()`. We can use this raw response to parse specific information contained in the response such a progress, time, percentage or other data. The operation will only check for changes in the `operation status`, but the operation-consumer will know better about all the data coming back from the server in the response.
+
+The next code demonstrates the use of `Poll()`.
+
+```cpp
+/**
+ * Consuming Operation<T> interface to Poll for LRO updates.
+ *
+ */
+auto operationResult = client.ApiReturnOperation(...);
+
+while (true)
+{
+    // Get a read-only ref to the Http raw response after polling. Poll will update the status.
+    auto& response = operationResult.Poll();
+
+    if(operationResult.IsDone()) {
+        // Operation completed.
+        break;
+    }
+
+    // Read the http headers in the response
+    for (auto& header : response.GetHeaders()) {
+        // ... consumer-code
+    }
+}
+
+// Check is the Operation has created the value `T`
+if (!operationResult.HasValue())
+{
+    // The Operation completed but it couldn't build the value `T`. Most likely the operation stage ended with error or it was cancelled.
+    throw std::runtime_error("Operation finished with status:" + operationResult.Status());
+}
+
+// At this point, the Operation is success and we can get its value.
+auto valueT = operationResult.Value();
+```
+
+### PollUntilDone
+
+The `PollUntilDone()` is a convenience method for repeatedly call `Poll()` until the status is updated to a _final stage_. A `Response<T>` will be returned wrapping the model `T` and the Http raw response from which it was created. This method is useful when we are not interested about the Http raw response for each polling request.
+
+The `PollUntilDone()` method requires an `std::chrono::milliseconds` input parameter which controls how much time to wait between calling `Poll`.
+
+In the code below, instead of manually polling for updates, the `PollUntilDone` method is called.
+
+```cpp
+/**
+ * Consuming Operation<T> interface. Calling PollUntilDone.
+ *
+ */
+auto operationResult = client.ApiReturnOperation(...);
+
+try
+{
+    // Update each second
+    auto response = operationResult.PollUntilDone(std::chrono::milliseconds(1000));
+
+    // No need to check for `HasValue` here. The Operation will throw if the value can't be created after completing.
+    auto valueT = response.ExtractValue();
```

Comment:
Were probably changing how this work a little bit.

Reply:
looking forward for it